modules/aleph/includes/aleph.availability.inc

```diff
@@ -94,21 +93,15 @@ function aleph_availability_holdings(array $provider_ids) {
         );
       }
 
-      // Count how many of the materials are reserved or available across sub-
-      // libraries.
-      if (!$item->isReservable()) {
-        $reserved_count++;
-      }
-      else {
-        $available_count++;
-      }
+      // Count how many of the materials are on shelf / available and how
+      // many are reservable.
+      $item->isAvailable() ? $available_count++ : $reserved_count++;
 
       // Provide data for each material from the provider ID.
       $data = array(
         'local_id' => $item->getId(),
         'available' => $available_count > 0,
-        'reservable' => $available_count > 0,
-        'show_reservation_button' => $available_count > 0,
+        'reservable' => count($items) > 0,
```

Comment:
This means that a record will always be reservable if any items exist. That does not seem right. 

How about checking if the number of reservable items is more than 0?

Reply:
Also: Is it intentional that you remove the `show_reservation_button` entry?

Reply:
`show_reservation_button` doesn't seem to have any functionality at all.
This is determined by `reservable`.